learn-data-structures-by-building-a-shopping-cart/index.js

```diff
@@ -1,97 +1,207 @@
+const shoppingCartContainer = document.getElementById(
+  "shopping-cart-container"
+);
+const productsContainer = document.getElementById("products-container");
+const dessertCards = document.getElementById("dessert-card-container");
+const shoppingCartBtn = document.getElementById("cart-btn");
+const clearShoppingCartBtn = document.getElementById("clear-cart-btn");
+const totalNumberOfItems = document.getElementById("total-items");
+const cartSubTotal = document.getElementById("subtotal");
+const cartTaxes = document.getElementById("taxes");
+const cartTotal = document.getElementById("total");
+const showHideCartSpan = document.getElementById("show-hide-cart");
+let isCartShowing = false;
+
+/**
+ *
+ * We can decide if we want to provide this products array for them or write steps where the campers
+ * will build it out themselves
+ *
+ */
 const products = [
   {
     id: 1,
-    name: "Vanilla buttercream cupcake",
-    price: 2.99,
-    category: "cupcake"
+    name: "Vanilla cupcakes (6 pack)",
+    price: 12.99,
+    category: "cupcake",
   },
   {
     id: 2,
     name: "French Macaroon",
     price: 3.99,
-    category: "macaroon"
+    category: "macaroon",
   },
   {
     id: 3,
-    name: "Fruit sprinkles cupcake",
+    name: "Pumpkin cupcake",
     price: 3.99,
-    category: "cupcake"
+    category: "cupcake",
   },
   {
     id: 4,
-    name: "Pink flower cupcake",
+    name: "Chocolate cupcake",
     price: 5.99,
-    category: "cupcake"
+    category: "cupcake",
   },
   {
     id: 5,
-    name: "Chocolate pretzel",
+    name: "Chocolate pretzels (4 pack)",
+    price: 10.99,
+    category: "pretzel",
+  },
+  {
+    id: 6,
+    name: "Strawberry ice cream",
+    price: 2.99,
+    category: "ice cream",
+  },
+  {
+    id: 7,
+    name: "Chocolate Macaroons (4 pack)",
+    price: 9.99,
+    category: "macaroon",
+  },
+  {
+    id: 8,
+    name: "Strawberry pretzel",
     price: 4.99,
-    category: "pretzel"
-  }
+    category: "pretzel",
+  },
+  {
+    id: 9,
+    name: "Butter Pecan ice cream",
+    price: 2.99,
+    category: "ice cream",
+  },
+  {
+    id: 10,
+    name: "Rocky Road ice cream",
+    price: 2.99,
+    category: "ice cream",
+  },
+  {
+    id: 11,
+    name: "Vanilla Macaroons (5 pack)",
+    price: 11.99,
+    category: "macaroon",
+  },
+  {
+    id: 12,
+    name: "Lemon cupcakes (4 pack)",
+    price: 12.99,
+    category: "cupcake",
+  },
 ];
 
+// another good instance to show how map and destructuring works
+products.map(
+  ({ name, id, price }) =>
+    (dessertCards.innerHTML += `
+    <div class="dessert-card">
+      <h2>${name}</h2>
+      <p class="dessert-price">$${price}</p>
+      <button 
+        id="${id}" 
+        class="btn add-to-cart-btn">Add to cart
+      </button>
+    </div>
+  `)
+);
```

Comment:
In case we want to show the product categories, too:

```js
products.map(
  ({ name, id, price, category }) =>
    (dessertCards.innerHTML += `
    <div class="dessert-card">
      <h2>${name}</h2>
      <p class="dessert-price">$${price}</p>
      <button 
        id="${id}" 
        class="btn add-to-cart-btn">Add to cart
      </button>
      <p>Category: ${category}</p>
    </div>
  `)
);
```

Reply:
I decided to add the category just before the cart button

<img width="381" alt="Screen Shot 2023-01-07 at 9 52 12 PM" src="https://user-images.githubusercontent.com/67210629/211182532-5341ecca-e96b-471d-b4c6-bd7bb5f321a5.png">